Build failure: pkgsi686Linux.python3Packages.myst-parser

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Make sure that you have a local clone of the Nixpkgs repo.

1. Change directory into your clone of the Nixpkgs repo by running this command:

    ```
    cd <path-to-repo>
    ```

1. Try to build the package by running this command:

    ```
    nix-build -A pkgsi686Linux.python3Packages.myst-parser
    ```

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/95gyx69kdd9q96fcdc4j4w79ysgaqp6x-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/tox.ini"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Setting SETUPTOOLS_SCM_PRETEND_VERSION to 2.7.0
Including all tracked files automatically
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
WARNING setuptools_scm.pyproject_reading toml section missing 'pyproject.toml does not contain a tool.setuptools_scm section'
Traceback (most recent call last):
  File "/nix/store/8kasna2zin48j7pasr0h84c5ig0yk5r7-python3.12-setuptools-scm-8.1.0/lib/python3.12/site-packages/setuptools_scm/_integration/pyproject_reading.py", line 36, in read_pyproject
    section = defn.get("tool", {})[tool_name]
              ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'setuptools_scm'
running bdist_wheel
running build
running build_py
creating build/lib/pytest_regressions
copying src/pytest_regressions/testing.py -> build/lib/pytest_regressions
copying src/pytest_regressions/plugin.py -> build/lib/pytest_regressions
copying src/pytest_regressions/__init__.py -> build/lib/pytest_regressions
copying src/pytest_regressions/image_regression.py -> build/lib/pytest_regressions
copying src/pytest_regressions/ndarrays_regression.py -> build/lib/pytest_regressions
copying src/pytest_regressions/common.py -> build/lib/pytest_regressions
copying src/pytest_regressions/data_regression.py -> build/lib/pytest_regressions
copying src/pytest_regressions/file_regression.py -> build/lib/pytest_regressions
copying src/pytest_regressions/dataframe_regression.py -> build/lib/pytest_regressions
copying src/pytest_regressions/num_regression.py -> build/lib/pytest_regressions
copying src/pytest_regressions/py.typed -> build/lib/pytest_regressions
installing to build/bdist.linux-i686/wheel
running install
running install_lib
creating build/bdist.linux-i686/wheel
creating build/bdist.linux-i686/wheel/pytest_regressions
copying build/lib/pytest_regressions/testing.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/data_regression.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/common.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/__init__.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/ndarrays_regression.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/plugin.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/num_regression.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/py.typed -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/dataframe_regression.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/image_regression.py -> build/bdist.linux-i686/wheel/./pytest_regressions
copying build/lib/pytest_regressions/file_regression.py -> build/bdist.linux-i686/wheel/./pytest_regressions
running install_egg_info
running egg_info
creating src/pytest_regressions.egg-info
writing src/pytest_regressions.egg-info/PKG-INFO
writing dependency_links to src/pytest_regressions.egg-info/dependency_links.txt
writing entry points to src/pytest_regressions.egg-info/entry_points.txt
writing requirements to src/pytest_regressions.egg-info/requires.txt
writing top-level names to src/pytest_regressions.egg-info/top_level.txt
writing manifest file 'src/pytest_regressions.egg-info/SOURCES.txt'
reading manifest file 'src/pytest_regressions.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'src/pytest_regressions.egg-info/SOURCES.txt'
Copying src/pytest_regressions.egg-info to build/bdist.linux-i686/wheel/./pytest_regressions-2.7.0-py3.12.egg-info
running install_scripts
creating build/bdist.linux-i686/wheel/pytest_regressions-2.7.0.dist-info/WHEEL
creating '/build/source/dist/.tmp-6mqgjivy/pytest_regressions-2.7.0-py3-none-any.whl' and adding 'build/bdist.linux-i686/wheel' to it
adding 'pytest_regressions/__init__.py'
adding 'pytest_regressions/common.py'
adding 'pytest_regressions/data_regression.py'
adding 'pytest_regressions/dataframe_regression.py'
adding 'pytest_regressions/file_regression.py'
adding 'pytest_regressions/image_regression.py'
adding 'pytest_regressions/ndarrays_regression.py'
adding 'pytest_regressions/num_regression.py'
adding 'pytest_regressions/plugin.py'
adding 'pytest_regressions/py.typed'
adding 'pytest_regressions/testing.py'
adding 'pytest_regressions-2.7.0.dist-info/LICENSE'
adding 'pytest_regressions-2.7.0.dist-info/METADATA'
adding 'pytest_regressions-2.7.0.dist-info/WHEEL'
adding 'pytest_regressions-2.7.0.dist-info/entry_points.txt'
adding 'pytest_regressions-2.7.0.dist-info/top_level.txt'
adding 'pytest_regressions-2.7.0.dist-info/RECORD'
removing build/bdist.linux-i686/wheel
Successfully built pytest_regressions-2.7.0-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for pytest_regressions-2.7.0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed pytest_regressions-2.7.0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/8mh5a70g3k7zqciavwwdxgcmima7zspz-python3.12-pytest-regressions-2.7.0
checking for references to /build/ in /nix/store/8mh5a70g3k7zqciavwwdxgcmima7zspz-python3.12-pytest-regressions-2.7.0...
patching script interpreter paths in /nix/store/8mh5a70g3k7zqciavwwdxgcmima7zspz-python3.12-pytest-regressions-2.7.0
stripping (with command strip and flags -S -p) in  /nix/store/8mh5a70g3k7zqciavwwdxgcmima7zspz-python3.12-pytest-regressions-2.7.0/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/sa2p710zz28s12yhmb2mnwamd01x4dmf-python3.12-pytest-regressions-2.7.0-dist
checking for references to /build/ in /nix/store/sa2p710zz28s12yhmb2mnwamd01x4dmf-python3.12-pytest-regressions-2.7.0-dist...
patching script interpreter paths in /nix/store/sa2p710zz28s12yhmb2mnwamd01x4dmf-python3.12-pytest-regressions-2.7.0-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: pytest_regressions pytest_regressions.plugin
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
pytest flags: -m pytest -W ignore::DeprecationWarning
============================= test session starts ==============================
platform linux -- Python 3.12.9, pytest-8.3.4, pluggy-1.5.0
rootdir: /build/source
configfile: tox.ini
plugins: regressions-2.7.0, datadir-1.5.0
collected 73 items                                                             

tests/test_data_regression.py .........                                  [ 12%]
tests/test_dataframe_regression.py ..............                        [ 31%]
tests/test_file_regression.py ....                                       [ 36%]
tests/test_filenames.py ...                                              [ 41%]
tests/test_grids.py ..                                                   [ 43%]
tests/test_image_regression.py ..                                        [ 46%]
tests/test_ndarrays_regression.py .F....................                 [ 76%]
tests/test_num_regression.py .................                           [100%]

=================================== FAILURES ===================================
_______________________________ test_common_case _______________________________

ndarrays_regression = <pytest_regressions.ndarrays_regression.NDArraysRegressionFixture object at 0xe200b390>
no_regen = None

    def test_common_case(ndarrays_regression, no_regen):
        # Most common case: Data is valid, is present and should pass
        data1 = np.full(5000, 1.1, dtype=float)
        data2 = np.arange(5000, dtype=int)
        ndarrays_regression.check({"data1": data1, "data2": data2})
    
        # Assertion error case 1: Data has one invalid place
        data1 = np.full(5000, 1.1, dtype=float)
        data2 = np.arange(5000, dtype=int)
        data1[500] += 0.1
        with pytest.raises(AssertionError) as excinfo:
            ndarrays_regression.check({"data1": data1, "data2": data2})
        obtained_error_msg = str(excinfo.value)
        expected = "\n".join(
            [
                "Values are not sufficiently close.",
                "To update values, use --force-regen option.",
            ]
        )
        assert expected in obtained_error_msg
        expected = "\n".join(
            [
                "data1:",
                "  Shape: (5000,)",
                "  Number of differences: 1 / 5000 (0.0%)",
                "  Individual errors:",
                "          Index              Obtained              Expected            Difference",
                "            500    1.2000000000000002                   1.1   0.10000000000000009",
            ]
        )
        assert expected in obtained_error_msg
    
        # Assertion error case 2: More than one invalid data
        data1 = np.full(5000, 1.1, dtype=float)
        data2 = np.arange(5000, dtype=int)
        data1[500] += 0.1
        data1[600] += 0.2
        data2[0] += 5
        data2[700:900] += 5
        with pytest.raises(AssertionError) as excinfo:
            ndarrays_regression.check({"data1": data1, "data2": data2})
        obtained_error_msg = str(excinfo.value)
        expected = "\n".join(
            [
                "Values are not sufficiently close.",
                "To update values, use --force-regen option.",
            ]
        )
        assert expected in obtained_error_msg
        expected = "\n".join(
            [
                "data1:",
                "  Shape: (5000,)",
                "  Number of differences: 2 / 5000 (0.0%)",
                "  Statistics are computed for differing elements only.",
                "  Stats for abs(obtained - expected):",
                "    Max:     0.19999999999999996",
                "    Mean:    0.15000000000000002",
                "    Median:  0.15000000000000002",
                "  Stats for abs(obtained - expected) / abs(expected):",
                "    Max:     0.18181818181818177",
                "    Mean:    0.13636363636363638",
                "    Median:  0.13636363636363638",
                "  Individual errors:",
                "          Index              Obtained              Expected            Difference",
                "            500    1.2000000000000002                   1.1   0.10000000000000009",
                "            600                   1.3                   1.1   0.19999999999999996",
            ]
        )
        assert expected in obtained_error_msg
        expected = "\n".join(
            [
                "data2:",
                "  Shape: (5000,)",
                "  Number of differences: 201 / 5000 (4.0%)",
                "  Statistics are computed for differing elements only.",
                "  Stats for abs(obtained - expected):",
                "    Max:     5",
                "    Mean:    5.0",
                "    Median:  5.0",
                "  Stats for abs(obtained - expected) / abs(expected):",
                "    Number of (differing) non-zero expected results: 200 / 201 (99.5%)",
                "    Relative errors are computed for the non-zero expected results.",
                "    Max:     0.007142857142857143",
                "    Mean:    0.006286830640674575",
                "    Median:  0.006253911138923655",
                "  Individual errors:",
                "    Only showing first 100 mismatches.",
                "          Index              Obtained              Expected            Difference",
                "              0                     5                     0                     5",
                "            700                   705                   700                     5",
                "            701                   706                   701                     5",
            ]
        )
>       assert expected in obtained_error_msg
E       AssertionError: assert 'data2:\n  Shape: (5000,)\n  Number of differences: 201 / 5000 (4.0%)\n  Statistics are computed for differing element...          700                     5\n            701                   706                   701                     5' in 'Values are not sufficiently close.\nTo update values, use --force-regen option.\n\ndata1:\n  Shape: (5000,)\n  Number...      797                     5\n            798                   803                   798                     5\n\n'

tests/test_ndarrays_regression.py:140: AssertionError
=========================== short test summary info ============================
FAILED tests/test_ndarrays_regression.py::test_common_case - AssertionError: assert 'data2:\n  Shape: (5000,)\n  Number of differences: ...
========================= 1 failed, 72 passed in 4.36s =========================
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.4, NixOS, 24.11 (Vicuna), 24.11.714876.5d7db4668d7a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos-24.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`


### Notify maintainers

@loicreynier

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pytest-regressions/default.nix
+++ b/pkgs/development/python-modules/pytest-regressions/default.nix
@@ -1,4 +1,5 @@
 {
+  stdenv,
   lib,
   buildPythonPackage,
   fetchFromGitHub,
@@ -61,6 +62,13 @@ buildPythonPackage rec {
 
   pytestFlags = [
     "-Wignore::DeprecationWarning"
+  ];
+
+  disabledTests = lib.optionals (stdenv.hostPlatform.isi686 || stdenv.hostPlatform.isBigEndian) [
+    # https://github.com/ESSS/pytest-regressions/issues/156
+    # i686-linux not listed in the report, but seems to have this issue as well
+    "test_different_data_types"
+    "test_common_case" # not listed in the issue, but fails after the above is skipped
   ];
 
   pythonImportsCheck = [

